src/JustSaying/AwsTools/MessageHandling/SnsMessagePublisher.cs

```diff
@@ -26,8 +29,9 @@ public class SnsMessagePublisher : IMessagePublisher, IInterrogable
         IMessageSerializationRegister serializationRegister,
         ILoggerFactory loggerFactory,
         IMessageSubjectProvider messageSubjectProvider,
-        Func<Exception, Message, bool> handleException = null)
-        : this(client, serializationRegister, loggerFactory, messageSubjectProvider, handleException)
+        Func<Exception, Message, bool> handleException = null,
+        Func<Exception, IReadOnlyCollection<Message>, bool> handleBatchException = null)
```

Comment:
This (and the change below) is a binary-breaking change as it adds new optional parameters.

If we can avoid needing a v8 that would be preferred, so instead you'd need to add a completely new constructor overload.